### PR TITLE
Add issue command

### PR DESCRIPTION
**Pull request by crvstalphua**
Created add issue command, with optional parameters

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@ src/test/data/sandbox/
 # MacOS custom attributes files created by Finder
 .DS_Store
 docs/_site/
+settings.gradle
```
Comment (ConradLew) [inline .gitignore:23]: Is this change intended?

File: src/main/java/seedu/address/logic/parser/IssueCliSyntax.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * Contains Command Line Interface (CLI) syntax definitions common to multiple commands
+ */
+public class IssueCliSyntax {
+
+    /* Prefix definitions */
+    public static final Prefix PREFIX_DESCRIPTION = new Prefix("dn/");
+    public static final Prefix PREFIX_DEADLINE = new Prefix("dl/");
+    public static final Prefix PREFIX_STATUS = new Prefix("s/");
+    public static final Prefix PREFIX_PRIORITY = new Prefix("py/");
+    public static final Prefix PREFIX_PROJECTID = new Prefix("pr/");
```
Comment (ConradLew) [inline src/main/java/seedu/address/logic/parser/IssueCliSyntax.java:13]: Consider refining the prefixes in the next iteration by using more intuitive prefix names

File: src/main/java/seedu/address/logic/parser/IssueParserUtil.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import seedu.address.model.issue.Priority;
+import seedu.address.model.issue.Status;
+import seedu.address.model.project.Project;
+import seedu.address.model.project.ProjectId;
+import seedu.address.model.project.UniqueProjectList;
+
+/**
+ * Contains utility methods used for parsing strings in the various *Parser classes.
+ */
+public class IssueParserUtil {
```
Comment (ConradLew) [inline src/main/java/seedu/address/logic/parser/IssueParserUtil.java:17]: Is it possible to use the overall `ParserUtil` classes? Otherwise parser methods such as `parseProject` may involve code duplication, and also necessitates the change in the overall structure where each entity class requires its own `ParserUtil` class e.g. `ProjectParserUtil`, `ClientParserUtil`

File: src/main/java/seedu/address/logic/parser/IssueParserUtil.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Parses a {@code String priority} into a {@code Priority}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code priority} is invalid.
+     */
+    public static Priority parsePriority(String priority) throws ParseException {
+        requireNonNull(priority);
+        String trimmedPriority = priority.trim().toUpperCase();
```
Comment (ConradLew) [inline src/main/java/seedu/address/logic/parser/IssueParserUtil.java:57]: Might have to be wary here of how the user inputs priority; it would be more intuitive for users to input `pr/0`, `pr/1` etc. rather than `pr/HIGH` which seems to be necessary here

File: src/main/java/seedu/address/logic/parser/IssueParserUtil.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /**
+     * Parses a {@code String status} into a {@code Status}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code status} is invalid.
+     */
+    public static Status parseStatus(String status) throws ParseException {
+        requireNonNull(status);
```
Comment (ConradLew) [inline src/main/java/seedu/address/logic/parser/IssueParserUtil.java:71]: There may be no need to parse the `Status`, the `Status` is not input by the user on default and simply defaults to false on creation

File: src/main/java/seedu/address/logic/parser/IssueParserUtil.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    /**
+     * Parses a {@code String projectId} into a {@code Project}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code projectId} is invalid.
+     */
+    public static Project parseProject(String projectId) throws ParseException {
+        requireNonNull(projectId);
+        String trimmedId = projectId.trim();
+        int trimmedIdInt = Integer.valueOf(trimmedId);
```
Comment (ConradLew) [inline src/main/java/seedu/address/logic/parser/IssueParserUtil.java:88]: Implementation here seems to differ from #79 , may want to standardize

File: src/main/java/seedu/address/model/issue/Issue.java [hunk 7]
```diff
@@ -31,10 +31,10 @@ public Issue(Description description, Deadline deadline, Priority priority,
         this.status = status;
         this.project = project;
         this.issueId = issueId;
+        this.project.getIssueList().add(this);
```
Comment (ConradLew) [inline src/main/java/seedu/address/model/issue/Issue.java:34]: good catch!

File: src/main/java/seedu/address/model/issue/Issue.java [hunk 8]
```diff
@@ -71,6 +71,12 @@ public Status getStatus() {
         return this.status;
     }
 
+    @Override
+    public String toString() {
```
Comment (ConradLew) [inline src/main/java/seedu/address/model/issue/Issue.java:75]: May want to create a separate method for the display implementation, since the `toString()` method is used to store the `IssueId` as JSON in the constructor of `JsonAdaptedIssue`

File: src/main/java/seedu/address/model/issue/UniqueIssueList.java [hunk 9]
```diff
@@ -68,6 +68,34 @@ public void setIssue(Issue target, Issue editedIssue) {
         internalList.set(index, editedIssue);
     }
 
+    public Issue getIssue(IssueId id) {
+        requireAllNonNull(id);
+        for (Issue i: internalList) {
+            if (i.getIssueId() == id) {
```
Comment (ConradLew) [inline src/main/java/seedu/address/model/issue/UniqueIssueList.java:74]: `==` may cause some unforeseen bugs later on, `.equals()` might be a better choice

File: src/main/java/seedu/address/model/project/UniqueProjectList.java [hunk 10]
```diff
@@ -69,6 +69,18 @@ public void setProject(Project target, Project editedProject) {
         internalList.set(index, editedProject);
     }
 
+    // TODO: temporary getter for project, can be deleted when alt method to retrieve project is implemented
+    public static Project getProject(int projectId) {
+        for (Project p : internalList) {
+            ProjectId id = new ProjectId(projectId);
+            if (p.getId() == id) {
```
Comment (ConradLew) [inline src/main/java/seedu/address/model/project/UniqueProjectList.java:76]: `==` may cause some unforeseen bugs later on, `.equals()` might be a better choice

File: src/main/java/seedu/address/ui/IssueCard.java [hunk 11]
```diff
@@ -43,10 +43,14 @@ public IssueCard(Issue issue, int displayedIndex) {
         super(FXML);
         this.issue = issue;
         id.setText(displayedIndex + ". ");
-        description.setText(issue.getDescription().toString() + issue.getId().uiRepresentation());
+        description.setText(issue.getDescription().toString() + " " + issue.getIssueId().uiRepresentation());
+        if (issue.getDeadline().isEmpty()) {
+            deadline.setText("No Deadline Set");
```
Comment (ConradLew) [inline src/main/java/seedu/address/ui/IssueCard.java:48]: Adding a method for the display string representation in each attribute class may be a better option, this way you can override the display method in the singleton class and will not have to do the checks here

**Review by ConradLew — COMMENTED** [review]
Overall, LGTM, minor possible improvements to the code. Do liase with @Donovan9617 and @Aishwarya-Hariharan-Iyer to see if the parsing of entities in the `ParserUtil` classes can be standardized!

File: .gitignore (hunk 1 above)
Comment (crvstalphua) [reply .gitignore:23]: sorry was an accident! undo it

**Review by ConradLew — APPROVED** [review]

**Review by Donovan9617 — APPROVED** [review]
